File: src/web/src/types/match.ts
import type {
  MatchStatus, ResultType, RoundType, VenueType,
  SurfaceType, BallColour, InningsType, BattingTeam,
  CommentaryCoverage,
} from './enums'

export interface Season {
  id: string
  name: string
  startDate: string
  endDate: string | null
}

export interface Tournament {
  id: string
  seasonId: string
  name: string
  format: 'ODI' | 'T20' | 'Other'
  oversPerSide: number
}

export interface Venue {
  id: string
  name: string
  city: string | null
  isMoraHomeGround: boolean
}

export interface Opponent {
  id: string
  name: string
  shortName: string
}

export interface Match {
  id: string
  tournamentId: string
  tournament: Tournament
  opponentId: string
  opponent: Opponent
  venueId: string | null
  venue: Venue | null
  matchDate: string
  scheduledOvers: number
  venueType: VenueType
  roundType: RoundType
  roundLabel: string | null
  tossHeld: boolean
  tossWinner: 'MORA' | 'OPPONENT' | null
  tossDecision: 'BAT' | 'FIELD' | null
  moraBattingFirst: boolean | null
  status: MatchStatus
  resultType: ResultType | null
  resultMargin: number | null
  resultMarginType: 'RUNS' | 'WICKETS' | null
  dlsApplied: boolean
  dlsTarget: number | null
  revisedOvers: number | null
  moraCaptainId: string | null
  moraWickeeperId: string | null
  playerOfMatchMoraId: string | null
  playerOfMatchName: string | null
  playerOfMatchTeam: 'MORA' | 'OPPONENT' | null
  surfaceType: SurfaceType
  ballColour: BallColour
  ballType: 'LEATHER' | 'TAPE' | 'TENNIS'
  notes: string | null
}

export interface Innings {
  id: string
  matchId: string
  inningsNumber: number
  inningsType: InningsType
  battingTeam: BattingTeam
  totalRuns: number
  totalWickets: number
  totalOversFaced: number
  extrasWides: number
  extrasNoBalls: number
  extrasLegByes: number
  extrasByes: number
  extrasPenalty: number
  hasDeliveryData: boolean
  commentaryCoverage: CommentaryCoverage
  moraWickeeperId: string | null
}

export interface FallOfWicket {
  id: string
  inningsId: string
  wicketNumber: number
  scoreAtFall: number
  overAtFall: number
  dismissedPlayerName: string
}

export interface Partnership {
  id: string
  inningsId: string
  wicketNumber: number
  moraBatter1Id: string | null
  moraBatter2Id: string | null
  oppBatter1Name: string | null
  oppBatter2Name: string | null
  runs: number
  balls: number
  batter1Runs: number
  batter2Runs: number
  unbroken: boolean
}

// Scorecard display
export interface MoraBattingLine {
  playerId: string
  playerName: string
  battingPosition: number
  runs: number
  ballsFaced: number
  fours: number
  sixes: number
  isNotOut: boolean
  minutesBatted: number | null
  dismissalType: string | null
  dismissedByName: string | null
  fielderName: string | null
}

export interface OpponentBattingLine {
  playerName: string
  battingPosition: number
  runs: number
  ballsFaced: number
  fours: number
  sixes: number
  isNotOut: boolean
  minutesBatted: number | null
  dismissalType: string | null
  dismissedByMoraPlayerName: string | null
  fielderName: string | null
}

export interface BowlingLine {
  playerName: string
  oversBowled: number
  maidens: number
  runsConceded: number
  wickets: number
  wides: number
  noBalls: number
}

File: src/api/MoraCricketHub.Domain/Entities/Match.cs
using System;
using System.Collections.Generic;
using System.Text;

using MoraCricketHub.Domain.Common;
using MoraCricketHub.Domain.Enums;

namespace MoraCricketHub.Domain.Entities;

public class Match : BaseEntity
{
    public Guid TournamentId { get; set; }
    public Tournament Tournament { get; set; } = null!;

    public Guid OpponentId { get; set; }
    public Opponent Opponent { get; set; } = null!;

    public Guid? VenueId { get; set; }
    public Venue? Venue { get; set; }

    public DateOnly MatchDate { get; set; }
    public int ScheduledOvers { get; set; } = 50;

    // Venue & conditions
    public VenueType VenueType { get; set; }
    public SurfaceType SurfaceType { get; set; } = SurfaceType.Matting;
    public BallColour BallColour { get; set; } = BallColour.Red;
    public BallType BallType { get; set; } = BallType.Leather;

    // Tournament round
    public RoundType RoundType { get; set; } = RoundType.FirstRound;
    public string? RoundLabel { get; set; }        // e.g. "Group B - Match 3"

    // Toss — all nullable when PreTossAbandoned
    public bool TossHeld { get; set; } = true;
    public string? TossWinner { get; set; }        // "MORA" | "OPPONENT" | null
    public string? TossDecision { get; set; }      // "BAT" | "FIELD" | null
    public bool? MoraBattingFirst { get; set; }

    // Result
    public MatchStatus Status { get; set; }
    public ResultType? ResultType { get; set; }
    public int? ResultMargin { get; set; }
    public string? ResultMarginType { get; set; }  // "RUNS" | "WICKETS"

    // DLS / Parabola
    public bool DlsApplied { get; set; } = false;
    public int? DlsTarget { get; set; }
    public int? RevisedOvers { get; set; }

    // Match officials
    public Guid? MoraCaptainId { get; set; }
    public Player? MoraCaptain { get; set; }

    public Guid? MoraWickeeperId { get; set; }
    public Player? MoraWickekeeper { get; set; }

    public string? OpponentCaptainName { get; set; }

    // Player of the Match
    public Guid? PlayerOfMatchMoraId { get; set; }
    public Player? PlayerOfMatchMora { get; set; }
    public string? PlayerOfMatchName { get; set; }
    public string? PlayerOfMatchTeam { get; set; } // "MORA" | "OPPONENT"

    public string? Notes { get; set; }

    // Navigation
    public ICollection<Innings> Innings { get; set; } = [];
    public ICollection<MatchSquad> Squad { get; set; } = [];
}
